### PR TITLE
trimText break on word boundary

### PR DESCRIPTION
**Pull request by bobvandevijver**
Currently, excerpt breaks words in the middle of it. This does not look nice. 

For example, with `"This is a text"|excerpt(12)` would become `This is a te...` With this PR. it will become `This is a ...`

When no space is found in the remaining text, there will be nothing extra cut.

### CODE REVIEW

**Comment by GawainLynch** [comment]
Thanks @bobvandevijver :+1: 

@bobdenotter (yep, this many Bobs could get tricky) if you're good with this change, I might send it down to `release/3.2` so the world gets it sooner than 3.4.

File: src/Helpers/Html.php [hunk 1]
```diff
@@ -26,7 +26,14 @@ public static function trimText($str, $desiredLength, $hellip = true)
         $str = trim(strip_tags($str));
 
         if (mb_strlen($str) > $desiredLength) {
-            $str = mb_substr($str, 0, $newLength) . $ellipseStr;
+            $nextChar = mb_substr($str, $newLength, 1);
+            $str = mb_substr($str, 0, $newLength);
+            if ($nextChar != ' ') {
+                if (false !== ($lastSpace = mb_strrpos($str, ' '))){
```
Comment (bobvandevijver) [inline src/Helpers/Html.php]: Updated commit fixing this :)

**Comment by bobdenotter** [comment]
I'd like it if there still was a cut off in there.. For example: 

`{{ "I suffer from hippopotomonstrosesquipedaliophobia"|trimtext(50) }}` -> "I suffer from …"

I'm fine with sneaking this into 3.2, but it'd be nice if there's a max to how many chars it might cut.

**Comment by bobvandevijver** [comment]
@bobdenotter Sure, what maximum would you propose?

**Comment by bobdenotter** [comment]
@bobvandevijver Maybe 8 or 10 max?

File: src/Helpers/Html.php [hunk 2]
```diff
@@ -26,7 +27,19 @@ public static function trimText($str, $desiredLength, $hellip = true)
...
         if (mb_strlen($str) > $desiredLength) {
-            $str = mb_substr($str, 0, $newLength) . $ellipseStr;
+            $nextChar = mb_substr($str, $newLength, 1);
+            $str = mb_substr($str, 0, $newLength);
+            if ($nextChar != ' ') {
+                if (false !== ($lastSpace = mb_strrpos($str, ' '))) {
+                    // Check for to long cutoff
+                    if (mb_strlen($str) - $lastSpace >= $cutOffCap) {
+                      // Trim the ellipse, as we do not want a space now
+                      return $str . trim($ellipseStr);
```
Comment (bobvandevijver) [inline src/Helpers/Html.php]: Fixed with updated commit.

**Comment by bobvandevijver** [comment]
@bobdenotter Here you are! Added a limit of 10, and you can adjust it to your needs.

**Comment by bobdenotter** [comment]
Looks good to me! Thanks, @bobvandevijver 👍

**Comment by GawainLynch** [comment]
@bobdenotter I'll take care of this in a second… but we might want to see if we can offer @bobvandevijver some assistance [with his hippopotomonstrosesquipedaliophobia](https://github.com/bolt/bolt/pull/5850/files#diff-6d68bc0dbdd4f150ec9ba3964735bc2fR26) :wink:

**Comment by GawainLynch** [comment]
@bobvandevijver  if you have a second… 
```
PHP Parse error:  syntax error, unexpected '}', expecting ',' or ')' in /home/travis/build/bolt/bolt/tests/phpunit/unit/Helper/HtmlTest.php on line 28
```

**Comment by bobvandevijver** [comment]
@GawainLynch Oh man, I though I committed the update correctly. Should be fixed now :)

**Comment by GawainLynch** [comment]
Zero stress mater … it's Sunday :smile:

**Comment by GawainLynch** [comment]
Merged into `release/3.2` vai 6900d315fa5df361f0104027b3f1996db15eaf1b

**Pull request closed without merging**